iceoryx_examples/icediscovery_in_c/README.md

```diff
@@ -0,0 +1,131 @@
+
+<!--[geoffrey][iceoryx_examples/icediscovery_in_c/iox_c_find_service.c][print function to be applied to search results]-->
+```c
+void printSearchResult(const iox_service_description_t service)
+{
+    printf(
+        "- Service: %s, Instance: %s, Event: %s\n", service.serviceString, service.instanceString, service.eventString);
+}
+```
+
+We can not only search for exactly matching services, but also for wildcards
+using `NULL`:
+
+<!--[geoffrey][iceoryx_examples/icediscovery_in_c/iox_c_find_service.c][search for all Radar services]-->
+```c
+iox_service_discovery_find_service_apply_callable(
+    serviceDiscovery, "Radar", NULL, NULL, printSearchResult, MessagingPattern_PUB_SUB);
+```
+
+With the above call we look for every `Radar` service with any instance and any
+event.
+
+Let's now search for all `Camera` services and store the results in an array:
+
+<!--[geoffrey][iceoryx_examples/icediscovery_in_c/iox_c_find_service.c][search for all Camera services]-->
+```c
+numberFoundServices = iox_service_discovery_find_service(serviceDiscovery,
+                                                         "Camera",
+                                                         NULL,
+                                                         NULL,
+                                                         searchResult,
+                                                         SEARCH_RESULT_CAPACITY,
+                                                         &missedServices,
+                                                         MessagingPattern_PUB_SUB);
+```
+
+`searchResult` is a `iox_service_description_t` array of size
+`SEARCH_RESULT_CAPACITY`. The found services are written into this array, but
+at maximum `SEARCH_RESULT_CAPACITY`. If the number of found services exceeds
+the array's capacity, the number of not stored services is written to
+`missedServices`. The number of stored services is returned. Since the
+`cameraPublishers` periodically offer/stop their services, you should see
+sometimes 5 `Camera` services and sometimes none.
+
+Finally, let's try out the third find service function. We search again for all
+`Camera` services but additionally count the front camera services:
+
+<!--[geoffrey][iceoryx_examples/icediscovery_in_c/iox_c_find_service.c][search for all front camera services]-->
+```c
+iox_service_discovery_find_service_apply_callable_with_context_data(
+    serviceDiscovery, "Camera", NULL, NULL, searchFrontDevices, &numberFrontCameras, MessagingPattern_PUB_SUB);
+```
+
+This function is quite similar to the first find service function, but we
+pass an additional argument:
+
+<!--[geoffrey][iceoryx_examples/icediscovery_in_c/iox_c_find_service.c][store number of front cameras]-->
+```c
+uint32_t numberFrontCameras = 0U;
+```
+
+We use this variable to store the number of front cameras and provide it as
+second parameter to the function which is applied to all found services:
```

Comment:
and provide a pointer to it as second argument `count` to the function `searchFrontDevices` which is applied